SSA/Projects/InstCombine/AliveAutoGenerated.lean

```diff
@@ -58,6 +59,78 @@ theorem alive_AddSub_1043  (w : Nat)   : alive_AddSub_1043_src w  ⊑ alive_AddS
   simp_alive_peephole
   apply bitvec_AddSub_1043
 
+-- Name: SimplifyDivRemOfSelect
+-- precondition: true
+/-
+%sel = select %c, %Y, 0
+%r = udiv %X, %sel
+  =>
+
+%r = udiv %X, %Y
+
+-/
+def alive_DivRemOfSelect_src (w : Nat) :=
+[mlir_icom (w)| {
+^bb0(%c: i1, %y : _, %x : _):
+  %c0 = "llvm.mlir.constant" () { value = 0 : _ } :() -> (_)
+  %v1 = "llvm.select" (%c,%y, %c0) : (i1, _, _) -> (_)
+  %v2 = "llvm.udiv"(%x, %v1) : (_, _) -> (_)
+  "llvm.return" (%v2) : (_) -> ()
+}]
+
+def alive_DivRemOfSelect_tgt (w : Nat) :=
+[mlir_icom (w)| {
+^bb0(%c: i1, %y : _, %x : _):
+  %v1 = "llvm.udiv" (%x,%y) : (_, _) -> (_)
+  "llvm.return" (%v1) : (_) -> ()
+}]
```

Comment:
I am surprised we add a new statement, instead of proofing one that is already in this file. Why was this not part of the statements that we autogenerated?

Reply:
The auto translation of select is broken, since select requires propagating the constraints of a variable of width ‘i1’. We do not have a full datafloe analysis in our alive translator, rather a best-effort implementation written by @goens as this was considered an acceptable time to effort trade off at the time.

Reply:
Sure, you could still add two theorems into a separate file, such that the changes to the auto-generated file are just two lines, no?

Reply:
I did not parse what you wrote. Are you suggesting we have a new file for this test case, as it is hand written? If so, I am happy to do so.

Reply:
This patch combines two rather unrelated changes:

1) adding a new theorem to AliveAutoGenerated without any explanation where it comes from

2) a proof of this theorem

I would have expected that you just proof a theorem that already existed. Why is this theorem not auto-generated by our alive extractor? Where does this come from, if not from our alive extractor?

Reply:
The issue is that we have manual edits to the `AliveStatements` file too, as I mentioned in Zulip. The two don't match because of this, that's why we (@bollu and I) had agreed that we'd re-generate `AliveStatements`. I'm thus also surprised that instead you are adding new theorems to the generated files. We should not be manually editing either of these two